pkg/translate/message.go

```diff
@@ -55,3 +58,72 @@ func (t *TranslateServiceServer) ListMessages(
 
 	return &translatev1.ListMessagesResponse{Messages: messagesSliceToProto(messages)}, nil
 }
+
+// ----------------------UpdateMessages-------------------------------
+
+type updateMessagesParams struct {
+	messages  *model.Messages
+	serviceID uuid.UUID
+}
+
+func parseUpdateMessagesRequestParams(req *translatev1.UpdateMessagesRequest) (*updateMessagesParams, error) {
+	serviceID, err := uuidFromProto(req.GetServiceId())
+	if err != nil {
+		return nil, fmt.Errorf("parse service_id: %w", err)
+	}
+
+	requestLanguageTag, err := language.Parse(req.GetMessages().Language)
+	if err != nil {
+		return nil, fmt.Errorf("parse language: %w", err)
+	}
+
+	messages := &model.Messages{
+		Language: requestLanguageTag,
+		Messages: make([]model.Message, len(req.GetMessages().GetMessages())),
+	}
+
+	for i, pbMsg := range req.GetMessages().GetMessages() {
+		messages.Messages[i] = model.Message{
+			ID:          pbMsg.GetId(),
+			Message:     pbMsg.GetMessage(),
+			Description: pbMsg.GetDescription(),
+			Fuzzy:       pbMsg.GetFuzzy(),
+		}
+	}
+
+	return &updateMessagesParams{serviceID: serviceID, messages: messages}, nil
+}
+
+func (u *updateMessagesParams) validate() error {
+	if u.serviceID == uuid.Nil {
+		return errors.New("'service_id' is required")
+	}
+
+	return nil
+}
+
+func (t *TranslateServiceServer) UpdateMessages(
+	ctx context.Context,
+	req *translatev1.UpdateMessagesRequest,
+) (*translatev1.UpdateMessagesResponse, error) {
+	params, err := parseUpdateMessagesRequestParams(req)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	if err = params.validate(); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	err = t.repo.SaveMessages(ctx, params.serviceID, params.messages)
```

Comment:
Shouldn't you check if messages already exist in DB before performing an update?
In a scenario, when db has no messages, then it will be created, which could be counterintuitive ? , because we already have createMessages endpoint

For me, it would be logically that we should not on updateRequest create resource if it does not exist. 
1

Reply:
The implementation would be similar to `CreateMessages`
```go
...
if len(msgs) < 1 {
  // nothing to update
}

// get the correct messages
...
t.repo.SaveMessages(with correct messages)

```

Reply:
@VladislavsPerkanuks advice `len(msgs) == 0` instead of `len(msgs) < 1` as it is easier to reason with. Where is `< 1` coming from?